### DISCUSSION TITLE
Merging output from different decision tables when one output is collect

### DISCUSSION THREAD

**Discussion by ShaamaHKazmi** [gorules/zen#90] [General]
Hi, I connected two decision tables to the output node. One of the decision tables returns a output based on the 'first' match to the output node and the second table 'collects' all the matches (i.e. a list) and returns it to the output node. I expected the output json would combine the two outputs into the resulting json, which it does when both of the output are selected based on 'first' match basis. But it's not the case if either one is a 'collect' and then it only returns output from the most recently connected edge.
Is their anyway to bypass this to return both the outputs even if one is a 'collect'?

**Comment by stefan-gorules** [comment]
Currently we do not support merging arrays with an object. This is the case when merging decision table collect with first hit policy.

To remedy this, usually you would put a function node next to a COLLECT decision table and map it via javascript e.g.

```js
/**
* @param {import('gorules').Input} input
* @param {{
*  moment: import('dayjs')
*  env: Record<string, any>
* }} helpers
*/
const handler = (input, { moment, env }) => {
  // Do some mapping 
  const sum = input.reduce((acc, item) => acc + item.amount, 0);

  return { sum };
}
```
